File: 백준/Silver/1748. 수 이어 쓰기 1/수 이어 쓰기 1.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int nLength = Integer.toString(n).length();
        long result = 0;
        
        for (int i = 1; i < nLength; i++) {
            result += 9 * (long) Math.pow(10, i - 1) * i;
        }
        result += (n - (int) Math.pow(10, nLength - 1) + 1) * nLength;
        System.out.println(result);
    }
}

File: 백준/Silver/1748. 수 이어 쓰기 1/수 이어 쓰기 1.py
n = int(input())
nLength = int(len(str(n)))
result = 0

for i in range(1, nLength):
    result += 9 * (10 ** (i - 1)) * i
result += (n - (10 ** (nLength - 1)) + 1) * nLength
print(result)